cst438_gradebook/src/main/java/com/cst438/controllers/GradeBookController.java

```diff
@@ -217,36 +159,95 @@ public void updateGradebook (@RequestBody GradebookDTO gradebook, @PathVariable(
 		
 	}
 	
-	// As an instructor, I can change the name of the assignment for my course.
-	@PutMapping("/gradebook/edit-name/{id}")
-	public void editAssignmentName (@RequestBody AssignmentDTO assignments, @PathVariable("id") Integer assignmentId ) {
-		
-		String email = "dwisneski@csumb.edu";  // user name (should be instructor's email) 
-		checkAssignment(assignmentId, email);  // check that user name matches instructor email of the course.
+	// As an instructor for a course , I can add a new assignment for my course.  
+	// The assignment has a name and a due date.
+	@PostMapping("/gradebook/add")
+	@Transactional
+	public AssignmentDTO addAssignment(@RequestBody AssignmentDTO assignmentDTO) {
+		Assignment duplicateAssignment = assignmentRepository.findById(assignmentDTO.assignmentId);
+			
+		if(duplicateAssignment == null) {
+			Assignment assignment = new Assignment();
+			assignment.setName(assignmentDTO.assignmentName);
+				
+			SimpleDateFormat formatter = new SimpleDateFormat("dd-MMM-yyyy", Locale.ENGLISH);
+			Date dueDate = null;
+			try {
+				dueDate = (Date) formatter.parse(assignmentDTO.dueDate);
+			} catch (ParseException e) {
+				// TODO Auto-generated catch block
+				e.printStackTrace();
+			}	
+			assignment.setDueDate(dueDate);
+				
+			Course course = courseRepository.findByCourse_id(assignmentDTO.courseId);
```

Comment:
Don't think it's necessary here but in real-life scenarios you would also check that the course id is valid.

Reply:
Good point, thanks.